src/Language/GraphQL/Draft/Parser.hs

```diff
@@ -53,6 +56,32 @@ runParser parser t =
 parseExecutableDoc :: Text -> Either Text AST.ExecutableDocument
 parseExecutableDoc = runParser executableDocument
 
+-- * Schema definition.
+
+-- | Parser for a 'AST.SchemaDefinition'.
+schemaDefinition :: Parser AST.SchemaDefinition
+schemaDefinition = tok "schema" *> braces (
+    AST.SchemaDefinition
+      <$> optional directives
+      <*> many1 rootOperationParser
+  )
+
+-- | Parser for a 'AST.RootOperationTypeDefinition'.
+rootOperationParser :: Parser AST.RootOperationTypeDefinition
+rootOperationParser =
+  AST.RootOperationTypeDefinition
+    <$> (operationTypeParser <* tok ":")
+    <*> (AST.NamedType <$> nameParser)
+
+-- | Parser for a 'AST.TypeSystemDefinition'.
+typeSystemDefinition :: Parser [AST.TypeSystemDefinition]
+typeSystemDefinition = whiteSpace *> (concat <$> many1 (
+      (\(AST.SchemaDocument d) -> AST.TypeSystemDefinitionType <$> d) <$> schemaDocument
+  <|> (: []) . AST.TypeSystemDefinitionSchema <$> schemaDefinition ))
+
+parseTypeSysDefinition :: Text -> Either Text [AST.TypeSystemDefinition]
+parseTypeSysDefinition = runParser typeSystemDefinition
+
```

Comment:
Personally I don't see the point of having these `parseX` definitions throughout the parser code. After all, we already export `runParser`. Let's leave this part out.